pkg/azure/azure.go

```diff
@@ -0,0 +1,55 @@
+package azure
+
+import (
+	"context"
+	"log/slog"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/grafana/cloudcost-exporter/pkg/provider"
+)
+
+const (
+	// subsystem = "azure"
+	collectTimeout time.Duration = 5 * time.Minute
```

Comment:
I really like this notion, but don't know if this is appropriate at a per provider level. How do you feel about making this an app level configuration that we can then pass to other providers as well?

Reply:
Good point, added it as a CLI arg and object on the top-level config